### PR TITLE
Call mixedref testing for all OpenJ9 builds

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Change the platform on OpenJ9 tests to be mixed.
This will run mixed testing using the existing test
jobs. This can be reverted once the default test
specs are mixed.

Add exception for 32bit (Windows JDK8) as these
builds are not mixed.

Extracted from adoptopenjdk/openjdk-build#2441
Resolves adoptopenjdk/openjdk-build#2426
Depends ibmruntimes/openj9-openjdk-jdk8#477

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

Coordinated merge with
ibmruntimes/openj9-openjdk-jdk11#382
ibmruntimes/openj9-openjdk-jdk16#23
ibmruntimes/openj9-openjdk-jdk8#478
ibmruntimes/openj9-openjdk-jdk#287
https://github.com/eclipse/openj9/pull/11951

### CODE REVIEW

**Review by karianna — APPROVED** [review]
LGTM

**Comment by gdams** [comment]
run tests

**Review by llxia — APPROVED** [review]
LGTM

**Comment by AdamBrousseau** [comment]
I should also mention, this needs to be a coordinated merge with
https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/382
https://github.com/ibmruntimes/openj9-openjdk-jdk16/pull/23
https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/478
https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/287
once
https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/477
lands.
Cc @pshipton

**Review by M-Davies — APPROVED** [review]

**Review by gdams — APPROVED** [review]

**Comment by pshipton** [comment]
@AdamBrousseau how feasible is it to modify this to only affect certain versions? We are currently blocked on jdk8, but could move jdk11 and 16 to mixedrefs builds. Since I don't know how long it will be until jdk8 is unblocked. it may be better to make the change for jdk11 and 16 to flush out any unexpected problems that will occur once mixedrefs is enabled, giving more time for these problems to be resolved. Not that I would do it on a Friday, but early next week. We can see if jdk8 makes any progress today.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (AdamBrousseau) [inline pipelines/build/common/openjdk_build_pipeline.groovy:278]: @pshipton 
```
if (buildConfig.VARIANT == "openj9" && versionInfo.major != 8) {
```
I'll leave it as-is until we decide.

**Comment by karianna** [comment]
@pshipton Were you able to unblock 8 or does @AdamBrousseau need to change this PR?  If it's a matter of a few days I'd suggest getting 8 fixed.

**Comment by gdams** [comment]
Looks like the issue was resolved in the upstream issue (the build is now passing). Hopefully, the J9 PR will be merged today and then we can follow up

**Comment by pshipton** [comment]
> Were you able to unblock 8

It should be unblocked now. I'll know in the morning based on the overnight test results if we are ready to proceed. If things looks good I'll put the PRs at OpenJ9 out for review and once they are merged this will need to be merged as well.

This PR and the linked PRs https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/382

> Looks like the issue was resolved in the upstream issue (the build is now passing).

I don't know anything about this, if it was a real problem it must have been unrelated to the blocking jdk8 issue I was referring to, as the fix was only merged within the last few hours.

**Comment by pshipton** [comment]
The overnight testing looks good, I'm discussing with a committer to make the switch to mixedrefs builds within the next few hours.

**Comment by pshipton** [comment]
Seems aarch64 is not yet ready to switch to mixedrefs.

**Comment by karianna** [comment]
@pshipton I'm setting this to draft so we don't accidentally land it before its deps are ready.

**Comment by pshipton** [comment]
I believe OpenJ9 is ready to proceed. It seems a bad idea to do this now, or on a Friday morning, in case it causes problems at Adopt over the weekend. We'll aim for Monday morning EST.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (pshipton) [inline pipelines/build/common/openjdk_build_pipeline.groovy:285]: As per https://github.com/eclipse/openj9/pull/11951#issuecomment-782229036, `_mixed` must not be added for 32-bit platforms.

**Comment by pshipton** [comment]
As an update is required to handle 32-bit platforms, not sure if this will be ready for merge on Monday. I'll wait for confirmation before we deliver the changes at OpenJ9.

File: pipelines/build/common/openjdk_build_pipeline.groovy
Comment (AdamBrousseau) [reply pipelines/build/common/openjdk_build_pipeline.groovy:286]: Added an exception to the `if` above for 32. Trying to test it internally but Adopt should test it.

**Comment by pshipton** [comment]
As per https://github.com/eclipse/openj9/pull/11951#issuecomment-782263376 we are discussing a different approach in TKG, so that this change is not needed at all.

**Comment by pshipton** [comment]
This change is no longer required as it's replaced by https://github.com/AdoptOpenJDK/TKG/pull/155

**Comment by gdams** [comment]
@pshipton closing thanks!

**Pull request closed without merging**
